InventoryAdminUi/view/adminhtml/ui_component/inventory_source_form.xml

```diff
@@ -94,6 +94,22 @@
                 </checkbox>
             </formElements>
         </field>
+        <field name="type_code" formElement="select" sortOrder="30">
```

Comment:
https://github.com/magento/inventory/blob/62f89e2f13f776cc6d30f97d4f2922e734ed3207/InventoryInStorePickupAdminUi/view/adminhtml/ui_component/inventory_source_form.xml#L71

It would be better to use something like `extension_attributes.type_code` (after reverting the `SourceDataProvider`)